.github/workflows/run-tests.yml

```diff
@@ -18,15 +18,46 @@ jobs:
           java-version: 11
           distribution: temurin
           cache: maven
+      - name: Check maven dependencies
+        run: mvn dependency:analyze --threads 2C --batch-mode -no-transfer-progress --update-snapshots -f ./dhis-2/pom.xml
 
-      - name: Test core
-        run: mvn clean install --threads 1C --batch-mode -no-transfer-progress --update-snapshots -f ./dhis-2/pom.xml -pl -dhis-web-embedded-jetty
+  check-formatting:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK 11
+        uses: actions/setup-java@v2
+        with:
+          java-version: 11
+          distribution: temurin
+      - name: Check formatting in core
+        run: mvn speedy-spotless:check --threads 2C --batch-mode --no-transfer-progress -q -f ./dhis-2/pom.xml
+      - name: Check formatting in web
+        run: mvn speedy-spotless:check --threads 2C --batch-mode --no-transfer-progress -q -f ./dhis-2/dhis-web/pom.xml
 
+  unit-test:
+    runs-on: ubuntu-latest
+    needs:
+      - check-formatting
+      - check-dependencies
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK 11
+        uses: actions/setup-java@v2
+        with:
+          java-version: 11
+          distribution: temurin
+          cache: maven
+      - name: Test core
+        run: mvn test --threads 2C --batch-mode -no-transfer-progress -f ./dhis-2/pom.xml -pl -dhis-web-embedded-jetty
```

Comment:
I think we should use `mvn package` at some point. If we don't then we might have a successful Github PR merged, to find that packaging fails on Jenkins which I am unsure how many devs looks at.

Not sure if package should be used in `Test core` and `Test dhis-web` or only in `Test dhis-web` (which I assume produces the dhis.war). Do we distribute any Java artifact (jar, war) from this repo other than the dhis.war?